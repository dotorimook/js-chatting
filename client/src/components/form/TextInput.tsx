import React, { FC } from 'react';
import IComponentProps from 'interfaces/IComponentProps';
import styled from 'styled-components';
import theme from 'style/theme';

interface ITextInput extends IComponentProps {
  type: string;
  value: string;
  name: string;
  placehoder?: string;
}

const Input = styled.input`
  font-size:1em;
  border:none;
  border-bottom:1px solid #666;
  padding:1em 1em;
  background-color:transparent;
  &::placeholder {
    color:${theme.gray9};
  }
`

const TextInput:FC<ITextInput> = (props:ITextInput) => <Input {...(props as any)} autoComplete='off'/>;

export default TextInput;