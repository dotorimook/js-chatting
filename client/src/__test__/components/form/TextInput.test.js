import React from 'react'
import 'setupTest';
import TextInput from 'components/form/TextInput';
import {shallow} from 'enzyme';

describe('TextInput Test', () => {
  let ui = null;
  const onChange = jest.fn();
  beforeAll(()=>{
    ui = shallow(<TextInput onChange={onChange} />);
  });
  
  it('matches snapshot', () => {
    expect(ui).toMatchSnapshot();
  });

  it('onChange called when change', () => {
    ui.simulate('change');
    expect(onChange).toHaveBeenCalled();
  });
});