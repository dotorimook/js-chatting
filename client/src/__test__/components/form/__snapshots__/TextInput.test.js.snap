// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Test matches snapshot 1`] = `
<styled.input
  autoComplete="off"
  onChange={[MockFunction]}
/>
`;
